File: SAAMSONSDISTRIBUTORS/DTOs/CartDto.cs
namespace SAAMSONSDISTRIBUTORS.Dtos
{
    public class AddToCartRequest
    {
        public int ProductId { get; set; }
        public int Quantity { get; set; } = 1;
        public string UserId { get; set; } = string.Empty; // required
    }

    public class UpdateCartQuantityRequest
    {
        public int Quantity { get; set; }
    }
}

File: SAAMSONSDISTRIBUTORSWEB/sam-sons-distributors/src/app/models/cart.dto.ts
export interface AddToCartRequest {
  userId: string;
  productId: number;
  quantity: number;
  // optional: clientId, discounts if added at add-time
}

export interface UpdateCartQuantityRequest {
  quantity: number; // 0 removes
}

export interface CheckoutRequest {
  userId: string;
  clientId: number;
  invoiceNumber?: string; // backend can generate if empty
}

export interface CheckoutResponse {
  message: string;
  invoiceNumber: string;
}
